### PR TITLE
fix: cache fixes

### PR DESCRIPTION
**Pull request by ronag**
## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: types/cache-interceptor.d.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ declare namespace CacheHandler {
   export type CacheMethods = 'GET' | 'HEAD' | 'OPTIONS' | 'TRACE'
 
   export interface CacheOptions {
-    store?: CacheStore
+    store: CacheStore
```
Comment (mcollina) [inline types/cache-interceptor.d.ts:9]: This is incorrect, we default to memory.

Comment (ronag) [reply types/cache-interceptor.d.ts:9]: Not according the cache handler. Then we have a different type issue

Comment (mcollina) [reply types/cache-interceptor.d.ts:9]: https://github.com/nodejs/undici/blob/a78147362135d7e5ea1617cef093fe5512a27ad9/lib/interceptor/cache.js#L90-L93 doesn't look that way.

Comment (ronag) [inline types/cache-interceptor.d.ts:9]:
```suggestion
    store?: CacheStore
```

Comment (mcollina) [reply types/cache-interceptor.d.ts:9]: This was not committed

Comment (ronag) [reply types/cache-interceptor.d.ts:9]: This is different. It's `CacheHandlerOptions` you are thinking of `CacheOptions`

Comment (mcollina) [reply types/cache-interceptor.d.ts:9]: Ah!

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**